### PR TITLE
fix(manage-issue-due-dates): paginate all issues without checking labels first

### PR DESCRIPTION
**Pull request by k-borley**
### :pencil: Description
- Fixes an issue with manage-issue-due-dates.  The paginateAllOpenIssues function was originally set up to return all open issues that had a priority label; unfortunately there is a limitation in the github REST API which will only return issues that have ALL of these labels. To remedy this the function now returns all open issues, which then get filtered in the manage issue due dates function using normal array filtering.

### :link: Related Issues
-

### CODE REVIEW

File: src/utils/paginate-prioritized-issues.ts [hunk 1]
```diff
@@ -14,19 +14,30 @@ limitations under the License.
...
-export const paginateAllOpenIssues = async (labels?: string, page = 1): Promise<IssueList> => {
+export const paginateAllPrioritizedIssues = async () => {
+  const prioritizedIssues: IssueList = [];
+  PRIORITY_LABELS.forEach(async label => {
+    const issues: IssueList = await paginateIssuesOfSpecificPriority(label);
+    // eslint-disable-next-line functional/immutable-data
+    prioritizedIssues.push(...issues);
+  });
+  return prioritizedIssues;
+};
```
Comment (danadajian) [inline src/utils/paginate-prioritized-issues.ts:27]: Let's use `bluebird.map` to keep consistent with the rest of the repo

File: README.md [hunk 2]
```diff
@@ -198,6 +192,12 @@ Additionally, the following parameters can be used for additional control over t
 
 -   Checks if the specifed GitHub user exists within an organization team
 
+### [manage-issue-due-dates](.github/workflows/manage-issue-due-dates.yml)
+
+-   Adds a comment listing the due date (based on SLA guidelines) to issues with a priority label attached
+-   Adds a &#x27;due soon&#x27; label to issues with a priority label that will become overdue in 7 days
+-   Adds an &#x27;overdue&#x27; label to issues with a priority label that are overdue
+
```
Comment (danadajian) [inline README.md:200]: Thanks for alphabetizing! 😄

File: src/utils/paginate-prioritized-issues.ts [hunk 3]
```diff
@@ -14,19 +14,24 @@ limitations under the License.
 import { IssueList } from '../types/github';
 import { octokit } from '../octokit';
 import { context } from '@actions/github';
+import { PRIORITY_LABELS } from '../constants';
+import { map } from 'bluebird';
 
-export const paginateAllOpenIssues = async (labels?: string, page = 1): Promise<IssueList> => {
+export const paginateAllPrioritizedIssues = async () =>
```
[the remark this answers is not in the document]
Comment (k-borley) [reply src/utils/paginate-prioritized-issues.ts:20]: paginateIssuesOfSpecificPriority returns an array of all the issues, and then map for some reason puts that array into another array, so I'm returning the actual result array here

File: src/utils/paginate-comments-on-issue.ts [hunk 4]
```diff
@@ -24,7 +24,7 @@ export const paginateAllCommentsOnIssue = async (issue_number: number, page = 1)
     page,
     ...context.repo
   });
-  if (!response.data.length) {
+  if (!response || !response.data?.length) {
```
Comment (danadajian) [inline src/utils/paginate-comments-on-issue.ts:27]:
```suggestion
  if (!response.data?.length) {
```

File: src/utils/paginate-prioritized-issues.ts [hunk 5]
```diff
@@ -14,19 +14,24 @@ limitations under the License.
...
+  (await map(PRIORITY_LABELS, async label => await paginateIssuesOfSpecificPriority(label))).filter(issues => issues.length > 0).flat();
```
Comment (danadajian) [inline src/utils/paginate-prioritized-issues.ts:21]:
```suggestion
  (await map(PRIORITY_LABELS, async label => await paginateIssuesOfSpecificPriority(label))).flat();
```

**Review by danadajian — APPROVED** [review]

**Pull request merged**